lib/ansible/modules/packaging/os/homebrew_cask.py

```diff
@@ -472,6 +484,25 @@ def _run(self):
         self.message = "You must select a cask to install."
         raise HomebrewCaskException(self.message)
 
+    # sudo_password fix ---------------------- {{{
+    def _run_command_with_sudo_password(self, cmd):
+        rc, out, err = '', '', ''
+
+        with tempfile.NamedTemporaryFile() as sudo_askpass_file:
+            sudo_askpass_file.write("#!/bin/sh\n\necho '" + self.sudo_password + "'\n")
```

Comment:
The file is opened in `w+b` mode by default. To make this Python2/3 compatible, write a string of bytes to the file:
```suggestion
            sudo_askpass_file.write(b"#!/bin/sh\n\necho '%s'\n" % to_bytes(self.sudo_password))
```

You will need to import `to_bytes`:

```python
from ansible.module_utils._text import to_bytes
```

It is faster to use `%` or `.format()` rather than string concatenation with the `+` operator. Concatenation results in `O(n**2)` operations  since each intermediate string is built whereas `%` and `.format()` result in `O(n)` operations.

In Python 3, bytes objects do not have the `.format()` method, so we need to use percent notation here.